operator/v1/types_insights.go

```diff
@@ -0,0 +1,113 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights
+	// +kubebuilder:validation:Required
+	// +required
+	Spec InsightsSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator
+	// +optional
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsStatus struct {
+	OperatorStatus `json:",inline"`
+	// GatheringStatus provides basic information about the last Insights gathering.
```

Comment:
Seeing `Gathering` everywhere doesn't really fit with the naming we use in APIs, how would you feel about this and others being just `GatherStatus` or `GatherFoo`? Does that detract from the meaning? 

We tend not to use adjectives in our API naming

Reply:
OK. I am not a native speaker. `GatheringStatus` sounds a little better to me, but I think we can go with `GatherStatus`.